src/storage/trades.cpp

```diff
@@ -0,0 +1,34 @@
+#include "trades.hpp"
+
+namespace raccoon {
+namespace storage {
+void
+TradeProcessor::process_incoming_match_(const Match& match)
+{
+    // nino , i didnt want to do it like this but ethan forced me
+    // pls dont annihilate me in pr
+    if (last_reset_ + std::chrono::seconds(1) < std::chrono::system_clock::now()) {
+        matches_.clear();
+        last_reset_ = std::chrono::system_clock::now();
+    }
+    matches_.push_back(match);
```

Comment:
Hmm

I feel like there's a smarter way to do this

We could do a [`SET`](https://redis.io/commands/set/) with expiry and then a [`SCAN`](https://redis.io/commands/scan/)

That way we don't even have to maintain the list of matches, and redis creates the rolling window for us.